lib/util/git/index.spec.ts

```diff
@@ -223,6 +223,12 @@ describe('util/git/index', () => {
 
   describe('mergeBranch(branchName)', () => {
     it('should perform a branch merge', async () => {
+      const repo = Git(GlobalConfig.get('localDir'));
+      await repo.checkout([
```

Comment:
🤔 it's currently still unclear to me why this is needed.

Reply:
Hi @viceice I'll try to explain the best way I can. So when I ran the original/unmodified tests, it complained: 

```
FAIL lib/util/git/index.spec.ts (35.928 s, 175 MB heap size)
  ● util/git/index › mergeBranch(branchName) › should perform a branch merge

    merge: renovate/future_branch - not something we can merge

      at Object.action (node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)
      at PluginStore.exec (node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)
      at node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40
      at GitExecutorChain.handleTaskData (node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)
      at GitExecutorChain.<anonymous> (node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)
      at fulfilled (node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)
          at runMicrotasks (<anonymous>)
```

The problem with the merge test seemed to be related to branch missing locally (Git can merge two branches if both of them are pulled): https://stackoverflow.com/questions/16862933/how-to-resolve-gits-not-something-we-can-merge-error

I had to modify the test setup to checkout `renovate/future_branch` from `origin/renovate/future_branch` before merging.

Reply:
Ah yes, that would normally done by the commit files function, which isn't called.